---
title: "urti2021"
output: html_document

---

          
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r cars}
library(vegan)
library(picante)
library(phyloseq)
library(plyr)
library(ggplot2)
library(reshape2)
library(patchwork) #拼图
library(cowplot)
library(ggpubr) #统计
library(GUniFrac)
library(microbiome)
library(ggthemes)
library(ggrepel)
library(tidyverse)
suppressWarnings(suppressMessages(library(amplicon)))
```


## make phyloseqs

```{r}

# setwd("qiime2out")
# 
# otu <- read.delim('feature-table.txt', row.names = 1, sep = '\t', stringsAsFactors = FALSE, check.names = FALSE)
# metadata <- read.table('metadata.txt', header = T, sep = '\t', row.names = 1)
# metadata <- metadata[colnames(otu),]
# tax<- read.table('tax.txt', header = T, sep = '\t', row.names = 1)
# tree<- read.tree('tree.nwk')
# repseqFile = "rep.fasta"
# rep.seq = Biostrings::readDNAStringSet(repseqFile)
# 
# otumat<- otu
# 
# OTU = otu_table(otumat, taxa_are_rows = TRUE)
# taxmat<- as.matrix(tax)
# TAX = tax_table(taxmat)
# sampledata<- sample_data(metadata)
# 
# 
# physeq0 = merge_phyloseq(OTU, TAX)
# physeq = merge_phyloseq(physeq0, sampledata, tree, rep.seq)
# physeq2 = merge_phyloseq(OTU, TAX, sampledata, tree, rep.seq)
# identical(physeq, physeq2)
# 
# sub_physeq <- subset_taxa(physeq, kindom == "k__Bacteria")
# prevdf = apply(X = otu_table(sub_physeq),
#                MARGIN = ifelse(taxa_are_rows(sub_physeq), yes = 1, no = 2),
#                FUN = function(x){sum(x > 0)})
# 
# physeq.great<- prune_taxa(names(which(prevdf >= 2)), sub_physeq)
# 
# set.seed(003)
# physeq.r = rarefy_even_depth(physeq.great, sample.size =20000, replace = TRUE)
# 
# physeq.rp <- prune_taxa(taxa_sums(physeq.r) > 0, physeq.r)
# 
# save(physea.rp, file = "physeq.rp.Rdata")
```


## Figure 2A

```{r}

load('physeq.rp.Rdata')
physeq.rp5<- subset_samples(physeq.rp, Gender != "NA")
physeq5<-  prune_taxa(taxa_sums(physeq.rp5) > 0, physeq.rp5)

GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="group")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "Gender", palette = c("black", "#aa3e35"),
          ellipse = TRUE,  alpha=0.75,
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

meta <- as.data.frame(meta(physeq5))
dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(123)
adonis2(dis_bray.f ~ meta$Gender, perm=999)



physeq.rp5<- subset_samples(physeq.rp, group != "NA")
physeq5<-  prune_taxa(taxa_sums(physeq.rp5) > 0, physeq.rp5)
meta <- meta(physeq5)
meta$FluA[which(meta$FluA >0)] <- "pos."
meta$FluB[which(meta$FluB >0)] <- "pos."
meta$RSV[which(meta$RSV >0)] <- "pos."
meta$RHV[which(meta$RHV >0)] <- "pos."
meta$RHV[which(meta$RHV != "pos.")] <- "neg."
meta$FluB[which(meta$FluB != "pos.")] <- "neg."
meta$FluA[which(meta$FluA != "pos.")] <- "neg."
meta$RSV[which(meta$RSV != "pos.")] <- "neg."
sample_data(physeq5)$groups <- meta$RHV
GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="groups")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "groups", palette = c("black", "#aa3e35"),
          ellipse = TRUE,  alpha=0.75,
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(123)
adonis2(dis_bray.f ~ meta$RHV, perm=999)


col5<- c("#824533", "#B79570","#CAD09E", "#FFCCB1", "#2B6B8E")

#FluA
physeq.rp5<- subset_samples(physeq.rp, group %in% c("CTL", "A"))
physeq5<-  prune_taxa(taxa_sums(physeq.rp5) > 0, physeq.rp5)

GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="group")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
p1 <- ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "group", palette = c("#824533", "#2B6B8E"),
          ellipse = TRUE,  
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

meta <- meta(physeq5)
dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(002)
adonis2(dis_bray.f ~ meta$group, perm=999)

#FluB
physeq.rp5<- subset_samples(physeq.rp, group %in% c("CTL", "B"))
physeq5<-  prune_taxa(taxa_sums(physeq.rp5) > 0, physeq.rp5)

GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="group")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
p2 <- ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "group", palette = c("#B79570", "#2B6B8E"),
          ellipse = TRUE,  
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

meta <- meta(physeq5)
dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(002)
adonis2(dis_bray.f ~ meta$group, perm=999)

#RSV
physeq.rp5<- subset_samples(physeq.rp, group %in% c("CTL", "S"))
physeq5<-  prune_taxa(taxa_sums(physeq.rp5) > 0, physeq.rp5)

GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="group")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
p3 <- ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "group", palette = c("#2B6B8E", "#CAD09E"),
          ellipse = TRUE,  
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

meta <- meta(physeq5)
dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(002)
adonis2(dis_bray.f ~ meta$group, perm=999)

# RHV
physeq.rp5<- subset_samples(physeq.rp, group %in% c("CTL", "H"))
physeq5<-  prune_taxa(taxa_sums(physeq.rp5) > 0, physeq.rp5)

GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="group")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
p4 <- ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "group", palette = c( "#2B6B8E", "#FFCCB1"),
          ellipse = TRUE,  
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

meta <- meta(physeq5)
dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(002)
adonis2(dis_bray.f ~ meta$group, perm=999)

p1 | p2 | p3 | p4

```



## Figure 2B-E

```{r}
############################################################################
## Figure 2B
lefse <- read.csv("dat/Figure_2_lefse.csv",header = T, sep = ',', row.names = 1)

lefse$genus <- rownames(lefse)
p1 <- ggplot(lefse,aes(x = reorder(genus,flua.lda),y=flua.lda,
              fill = flua.group,
              label = flua.lda))+
    geom_col(show.legend = FALSE) +
    coord_flip() +
    xlab('Gene Name') +
    ggtitle('Expression level of significant changed genes')+
    theme_clean()+
    scale_fill_manual(values = c( "black", "#824533", "#2B6B8E"))+
    labs(title= "FluA vs Health", x="", y="LDA score(log10)")

p2 <- ggplot(lefse,aes(x = reorder(genus,flua.lda),y=flub.lda,
              fill = flub.group,
              label = flub.lda))+
    geom_col(show.legend = FALSE) +
    coord_flip() +
    xlab('Gene Name') +
    ggtitle('Expression level of significant changed genes')+
    theme_clean()+
    scale_fill_manual(values = c( "black", "#B79570", "#2B6B8E"))+
    theme(axis.text.y = element_blank())+
    labs(title= "FluB vs Health", x="", y="LDA score(log10)")

p3 <- ggplot(lefse,aes(x = reorder(genus,flua.lda),y=rsv.lda,
              fill = rsv.group,
              label = rsv.lda))+
    geom_col(show.legend = FALSE) +
    coord_flip() +
    xlab('Gene Name') +
    ggtitle('Expression level of significant changed genes')+
    theme_clean()+
    scale_fill_manual(values = c( "black", "#2B6B8E", "#CAD09E"))+
    theme(axis.text.y = element_blank())+
    labs(title= "RSV vs Health", x="", y="LDA score(log10)")

p4 <- ggplot(lefse,aes(x = reorder(genus,flua.lda),y=rhv.lda,
              fill = rhv.group,
              label = rhv.lda))+
    geom_col(show.legend = FALSE) +
    coord_flip() +
    xlab('Gene Name') +
    ggtitle('Expression level of significant changed genes')+
    theme_clean()+
    scale_fill_manual(values = c( "black", "#2B6B8E", "#FFCCB1"))+
    theme(axis.text.y = element_blank())+
    labs(title= "HRV vs Health", x="", y="LDA score(log10)")

p1 | p2 | p3 | p4

############################################################################
## Figure 2C

otus1$group <- factor(otus1$group,levels=c("A", "B", "S", "H", "CTL"))
my_comparisons5 <- list(c("CTL", "H"), c("CTL", "S"), c("CTL", "B"), c("A", "CTL"))

p1 <- ggplot(data=otus1, aes(x = group, y = 100*g__Veillonella, group=group)) +
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour = "grey") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="grey") +
    geom_jitter(width=0.2, size=1.5, alpha = 0.75) +
    scale_fill_manual(values=col5) +
    theme(title=element_text(size=12))+
    stat_compare_means(paired = FALSE, color="black")+
    stat_compare_means(comparisons = my_comparisons5, paired = FALSE, color="black")+
    theme_classic()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(plot.title = element_text(hjust = 0.5))+
    labs(title= "Veillonella", y="Realative Abundance", x="")
p.bonferroni = c(0.013, 0.00036, 0.032, 0.0018)*4 


############################################################################
## Figure 2D

p2 <- ggplot(data=otus1, aes(x = group, y = 100*g__Granulicatella, group=group)) +
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour = "grey") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="grey") +
    geom_jitter(width=0.2, size=1.5, alpha = 0.75) +
    scale_fill_manual(values=col5) +
    theme(title=element_text(size=12))+
    stat_compare_means(paired = FALSE, color="black")+
    stat_compare_means(comparisons = my_comparisons5, paired = FALSE, color="black")+
    theme_classic()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(plot.title = element_text(hjust = 0.5))+
    labs(title= "Granulicatella", y="Realative Abundance", x="")
p.bonferroni <- c(0.054, 2e-5, 0.00232, 0.00058)*4 

############################################################################
## Figure 2E


data <- -lefse[,c(2,6,10,14)]
data1 <- (data-min(data))/(max(data)-min(data))
colnames(data1) <- c("FluA", "FluB", "RSV", "HRV")
library(ggbiplot)
win.pca <- prcomp(data1, scale. = TRUE)
ggbiplot(win.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = FALSE) +
    scale_color_discrete(name = '') +
    theme(legend.direction = 'horizontal', legend.position = 'top')+
    theme_classic()+
    geom_text_repel(aes(label=rownames(data1)))

```


## Figure 3A

```{r}
library(pROC) 
library(randomForest) 
library(e1071)
library(glmnet)

physeq.rpa<- subset_samples(physeq.rp, group %in% c("A", "B", "S", "H", "AB", "ABS", "AS", "BH", "BS", "CTL"))

physeq.rpa<-  prune_taxa(taxa_sums(physeq.rpa) > 0, physeq.rpa)
physeq10 = physeq.rpa

physeq1 <- subset_samples(physeq10, group %in% c("A", "CTL"))
data <- as.data.frame(t(physeq1@otu_table@.Data))
colnames(data) <- paste0(as.data.frame(physeq1@tax_table@.Data)$Taxon, colnames(data))
colnames(data) <- make.names(colnames(data))
data1 <- data

data1$group <- as.factor(as.data.frame(meta(physeq1))$group)
set.seed(123)
rf.fit<- randomForest(group ~ ., data = data1, importance = TRUE, )
A <- plot.roc(data1$group, 
              rf.fit$votes[,1], 
              percent=TRUE,
              xlab="False Positive Percentage", 
              ylab="True Postive Percentage", col="#377eb8", lwd=4, print.auc=TRUE)

physeq1 <- subset_samples(physeq10, group %in% c("B", "CTL"))
data <- as.data.frame(t(physeq1@otu_table@.Data))
colnames(data) <- paste0(as.data.frame(physeq1@tax_table@.Data)$Taxon, colnames(data))
colnames(data) <- make.names(colnames(data))
data1 <- data

data1$group <- as.factor(as.data.frame(meta(physeq1))$group)
set.seed(123)
rf.fit<- randomForest(group ~ ., data = data1, importance = TRUE, )
B <- plot.roc(data1$group, 
              rf.fit$votes[,1], 
              percent=TRUE,
              xlab="False Positive Percentage", 
              ylab="True Postive Percentage", col="#377eb8", lwd=4, print.auc=TRUE)

physeq1 <- subset_samples(physeq10, group %in% c("S", "CTL"))
data <- as.data.frame(t(physeq1@otu_table@.Data))
colnames(data) <- paste0(as.data.frame(physeq1@tax_table@.Data)$Taxon, colnames(data))
colnames(data) <- make.names(colnames(data))
data1 <- data

data1$group <- as.factor(as.data.frame(meta(physeq1))$group)
set.seed(123)
rf.fit<- randomForest(group ~ ., data = data1, importance = TRUE, )
S <- plot.roc(data1$group, 
              rf.fit$votes[,1], 
              percent=TRUE,
              xlab="False Positive Percentage", 
              ylab="True Postive Percentage", col="#377eb8", lwd=4, print.auc=TRUE)

physeq1 <- subset_samples(physeq10, group %in% c("H", "CTL"))
data <- as.data.frame(t(physeq1@otu_table@.Data))
colnames(data) <- paste0(as.data.frame(physeq1@tax_table@.Data)$Taxon, colnames(data))
colnames(data) <- make.names(colnames(data))
data1 <- data

data1$group <- as.factor(as.data.frame(meta(physeq1))$group)
set.seed(123)
rf.fit<- randomForest(group ~ ., data = data1, importance = TRUE, )
H <- plot.roc(data1$group, 
              rf.fit$votes[,1], 
              percent=TRUE,
              xlab="False Positive Percentage", 
              ylab="True Postive Percentage", col="#377eb8", lwd=4, print.auc=TRUE)

physeq1 <- subset_samples(physeq10, group1 %in% c("infection", "control"))
data <- as.data.frame(t(physeq1@otu_table@.Data))
colnames(data) <- paste0(as.data.frame(physeq1@tax_table@.Data)$Taxon, colnames(data))
colnames(data) <- make.names(colnames(data))
data1 <- data

data1$group <- as.factor(as.data.frame(meta(physeq1))$group1)
set.seed(123)
rf.fit<- randomForest(group ~ ., data = data1, importance = TRUE, )
i <- plot.roc(data1$group, 
              rf.fit$votes[,1], 
              percent=TRUE,
              xlab="False Positive Percentage", 
              ylab="True Postive Percentage", col="#377eb8", lwd=4, print.auc=TRUE)

plot(A, time = 3, col="#95641f", lwd=2, title = "")
plot(B, time = 3, col="#c4934e", lwd=2, add = T)
plot(S, time = 3, col="#808000", lwd=2, add = T)
plot(H, time = 3, col="#f2b77b", lwd=2, add = T)
plot(i, time = 3, col="#054e61", lwd=2, add = T)
legend("bottomright",
       c(paste0("FluA: ",round(A[["auc"]],2)), 
         paste0("FluB: ",round(B[["auc"]],2)), 
         paste0("RSV: ",round(S[["auc"]],2)),
         paste0("HRV: ",round(H[["auc"]],2)),
         paste0("infection: ",round(i[["auc"]],2))
       ),
       col=c("#95641f", "#c4934e", "#808000","#f2b77b","#054e61"),
       lty=1, lwd=2,bty = "n")

```


## Figure 3B-C

```{r}
#######################################################################
## Figure 3B

roc <- read.csv("dat/Figure_3_roc.csv",header = T, sep = ',', row.names = 1)
roc1 <- melt(roc)
col12<- c("#824533", "#B79570","#CAD09E", "#FFCCB1", "#2B6B8E")
ggplot(roc1, aes(x=variable, y=value, fill=variable)) +
    geom_bar(stat="summary", fun=mean, position="dodge") +
    stat_summary(geom = "errorbar",fun.data = 'mean_sd', width = 0.3)+
    scale_fill_manual(values=col12) +
    theme_classic()+
    geom_jitter(data = roc1,  aes(x=variable, y=value), width=0.2, size=1.5, alpha = 0.75)+
    coord_cartesian( ylim = c(60, 100))+
    labs( y="AUC", x="")


#######################################################################
## Figure 3C

imp <- read.csv("dat/Figure_3_imp.csv",header = T, sep = ',')
data <- as.data.frame(aggregate(imp$MeanDecreaseAccuracy, by=list(type=imp$Group_type,imp$MeanDecreaseAccuracy), mean))

ss <- subset(imp, out_id %in% tax1)
tax <- physeq.rpa@tax_table@.Data
ss$taxon <- as.data.frame(tax[ss$out_id,])$Taxon
ss$phylum <- as.data.frame(tax[ss$out_id,])$phylum

ss$out_id <- factor(ss$out_id,levels=rev(tax1))
ss$Group_type <- factor(ss$Group_type,levels= c("A", "B", "S", "H", "CTL"))
ggplot(ss, aes(y=out_id, x=MeanDecreaseAccuracy)) +
    geom_bar(stat="summary", fun=mean, position="dodge") +
    stat_summary(geom = "errorbar",fun.data = 'mean_sd', width = 0.2)+
    theme_classic2()+
    geom_jitter(width=0.03, size=1, alpha = 0.3)+
    labs( y="Taxon", x="")+
    facet_grid(.~Group_type)+
    scale_y_discrete(labels = rev(as.data.frame(tax[tax1,])$Taxon)) 

```


## Figure 4A

```{r}
s.sh <- plot_richness(physeq.rp, "group",  measures=NULL)
s.data <- s.sh$data

s.data$BMI[which(s.data$BMI > 21.22)] = "BMI > 21.22"
s.data$BMI[which(s.data$BMI <= 21.22)] = "BMI ≤ 21.22"
s.data$Age[which(s.data$Age > 29)] = "Age > 29"
s.data$Age[which(s.data$Age < 9)] = "Age ≤ 29"
s.data$Age[which(s.data$Age <= 29)] = "Age ≤ 29"
s.data$FluA[which(s.data$FluA > 0)] = "FluA pos."
s.data$FluA[which(s.data$FluA == "0")] = "FluA neg"
s.data$FluB[which(s.data$FluB > 0)] = "FluB pos."
s.data$FluB[which(s.data$FluB == "0")] = "FluB neg"
s.data$RSV[which(s.data$RSV > 0)] = "RSV pos."
s.data$RSV[which(s.data$RSV == "0")] = "RSV neg"
s.data$RHV[which(s.data$RHV > 0)] = "HRV pos."
s.data$RHV[which(s.data$RHV == "0")] = "HRV neg"
s.data1 <- subset(s.data, variable %in% c("Shannon"))
dat <- melt(s.data1[,c(1:7,9,13)], measure.vars = colnames(s.data1[,c(1:7,9)]))

colnames(dat) <- c("value", "type", "group")
dat$type <- factor(dat$type, levels = c("Gender", "Age", "BMI", "FluA", "FluB", "RSV", "RHV", "group1"))
ggviolin(dat, "group", "value", fill = "group",
         add = "boxplot", add.params = list(fill = "white"))+
    facet_wrap(.~type, scales = "free", nrow=1)+
    theme_classic()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(plot.title = element_text(hjust = 0.5))

compare_means(value~group, subset(dat, type == "Gender"))
compare_means(value~group, subset(dat, type == "Age"))
compare_means(value~group, subset(dat, type == "BMI"))
compare_means(value~group, subset(dat, type == "FluA"))
compare_means(value~group, subset(dat, type == "FluB"))
compare_means(value~group, subset(dat, type == "RSV"))
compare_means(value~group, subset(dat, type == "RHV"))
compare_means(value~group, subset(dat, type == "group1"))

```


## Figure 4B & Figure S3

```{r}
physeq.rp5<- subset_samples(physeq.rp, Gender != "NA")
physeq5<-  prune_taxa(taxa_sums(physeq.rp5) > 0, physeq.rp5)

GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="group")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
p1 <- ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "Gender", palette = c("black", "#aa3e35"),
          ellipse = TRUE,  alpha=0.75,
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

meta <- as.data.frame(meta(physeq5))
dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(123)
adonis2(dis_bray.f ~ meta$Gender, perm=999)


physeq.rp5<- subset_samples(physeq.rp, Age != "NA")
physeq5<-  prune_taxa(taxa_sums(physeq.rp5) > 0, physeq.rp5)
meta <- meta(physeq5)
meta$Age[which(meta$Age > 29)] = "Age > 29"
meta$Age[which(meta$Age < 9)] = "Age ≤ 29"
meta$Age[which(meta$Age <= 29)] = "Age ≤ 29"
sample_data(physeq5)$Age <- meta$Age
GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="groups")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
p2 <- ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "Age", palette = c("black", "#aa3e35"),
          ellipse = TRUE,  alpha=0.75,
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(123)
adonis2(dis_bray.f ~  meta$Age, perm=999)


physeq.rp5<- subset_samples(physeq.rp, BMI != "NA")
physeq5<-  prune_taxa(taxa_sums(physeq.rp5) > 0, physeq.rp5)
meta <- meta(physeq5)
meta$BMI[which(meta$BMI > 21.22)] = "BMI > 21.22"
meta$BMI[which(meta$BMI <= 21.22)] = "BMI ≤ 21.22"
sample_data(physeq5)$BMI <- meta$BMI
GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="groups")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
p3 <- ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "BMI", palette = c("black", "#aa3e35"),
          ellipse = TRUE,  alpha=0.75,
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(123)
adonis2(dis_bray.f ~  meta$BMI, perm=999)


physeq.rp5<- subset_samples(physeq.rp, group != "NA")
physeq5<-  prune_taxa(taxa_sums(physeq.rp5) > 0, physeq.rp5)
meta <- meta(physeq5)
meta$FluA[which(meta$FluA >0)] <- "pos."
meta$FluB[which(meta$FluB >0)] <- "pos."
meta$RSV[which(meta$RSV >0)] <- "pos."
meta$RHV[which(meta$RHV >0)] <- "pos."
meta$RHV[which(meta$RHV != "pos.")] <- "neg."
meta$FluB[which(meta$FluB != "pos.")] <- "neg."
meta$FluA[which(meta$FluA != "pos.")] <- "neg."
meta$RSV[which(meta$RSV != "pos.")] <- "neg."

sample_data(physeq5)$groups <- meta$FluA
GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="groups")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
p5 <- ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "groups", palette = c("black", "#aa3e35"),
          ellipse = TRUE,  alpha=0.75,
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(123)
adonis2(dis_bray.f ~ meta$FluA, perm=999)


sample_data(physeq5)$groups <- meta$FluB
GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="groups")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
p6 <- ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "groups", palette = c("black", "#aa3e35"),
          ellipse = TRUE,  alpha=0.75,
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(123)
adonis2(dis_bray.f ~ meta$FluB, perm=999)

sample_data(physeq5)$groups <- meta$RSV
GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="groups")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
p7 <- ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "groups", palette = c("black", "#aa3e35"),
          ellipse = TRUE,  alpha=0.75,
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(123)
adonis2(dis_bray.f ~ meta$RSV, perm=999)

sample_data(physeq5)$groups <- meta$RHV
GP.ord <- ordinate(physeq5, "PCoA", "bray")
data1<- plot_ordination(physeq5, GP.ord, color="groups")
c(data1[["labels"]][["x"]], data1[["labels"]][["y"]])
data2 <- data1$data
p8 <- ggscatter(data2, x= "Axis.1", y = "Axis.2", 
          color = "groups", palette = c("black", "#aa3e35"),
          ellipse = TRUE,  alpha=0.75,
          mean.point = FALSE, star.plot = FALSE,  
          ellipse.level = 0.3,  
          ggtheme = theme_minimal(),
          rug = FALSE)+
    theme(axis.title.x = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme(axis.title.y = element_text(size = 16,
                                      face = "bold", 
                                      vjust = 0.5, 
                                      hjust = 0.5))+
    theme_classic()+
    xlab(data1[["labels"]][["x"]]) + ylab(data1[["labels"]][["y"]])

dis_bray.f<- phyloseq::distance(physeq5, "bray")
set.seed(123)
adonis2(dis_bray.f ~ meta$RHV, perm=999)

(p1 |p2 | p3)/ (p4 |p5 | p6)
(p7 | p8 )

```


## Figure 4C

The source data for importing Cytoscape and the Cytoscape output files are named "Figure 4C.csv" and "Figure 5C.cys"respectively in "dat" file.

## Figure S4-5

```{r}

fs45 <- read.csv("dat/Figure_s4_5.csv",header = T, sep = ',')

my_comparisons <- list(c("FluA", "Health"), c("FluB", "Health"), c("FluAB", "Health"))
fs45$group <- factor(fs45$group, levels = c("FluA", "FluB", "FluAB", "Health"))
ggplot(data=fs45, aes(x = group, y = arg, group=group)) +
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour = "grey") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="grey") +
    geom_jitter(width=0.2, size=1.5, alpha = 0.75) +
    scale_fill_manual(values= c("#824533", "#b79570", "#161645", "#2b6b8e")) +
    theme(title=element_text(size=12))+
    stat_compare_means(paired = FALSE, color="black")+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE, color="black")+
    theme_classic()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(plot.title = element_text(hjust = 0.5))+
    labs(title= "", y="ARGs counts", x="")


ggplot(data=fs45, aes(x = group, y = all, group=group)) +
    geom_boxplot(aes(x=group, fill=group), notch=FALSE, outlier.colour = "grey") +
    stat_summary(aes(x=group), fun.y=mean, geom="point", color="grey") +
    geom_jitter(width=0.2, size=1.5, alpha = 0.75) +
    scale_fill_manual(values= c("#824533", "#b79570", "#161645", "#2b6b8e")) +
    theme(title=element_text(size=12))+
    stat_compare_means(paired = FALSE, color="black")+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE, color="black")+
    theme_classic()+ 
    theme(axis.text.x = element_text(size=14),axis.text.y = element_text(size=14)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
    theme(legend.title = element_blank())+ 
    theme(legend.position='none')+
    theme(plot.title = element_text(hjust = 0.5))+
    labs(title= "", y="Realative Abundance (%)", x="")



```


## Figure 5A-C

```{r}
########################################################################
## Figure 5A


grid <- read.csv("dat/Figure_5_grid.csv",header = T, sep = ',')

names1 <- c("Streptococcus pneumoniae", "Mycoplasma pneumoniae", "Haemophilus influenzae", "Klebsiella pneumoniae", "Pseudomonas aeruginosa", "Staphylococcus aureus", "Chlamydia pneumoniae", "Legionella pneumophila", "Escherichia coli", "Treponema denticola", "Porphyromonas gingivalis", "Haemophilus sputorum","Staphylococcus epidermidis", "Fusobacterium nucleatum", "Treponema medium")



grid$species[which(grid$species %in% names1)] = "Pathogens"
grid$species[which(grid$species != "Pathogens")] = "Commesals"
grid$groups <- paste(grid$group, grid$species)
grid$GRiD[which(grid$GRiD > 5)]  = 5

grid$groups <- factor(grid$groups, levels = c("FluA Pathogens", "FluA Commesals", "FluB Pathogens", "FluB Commesals", "FluAB Pathogens", "FluAB Commesals", "Health Pathogens", "Health Commesals"))
my_comparisons <- list(c("FluA Pathogens", "FluA Commesals"), c("FluB Pathogens", "FluB Commesals"), c("FluAB Pathogens", "FluAB Commesals"), c("Health Pathogens", "Health Commesals"), c("FluA Pathogens", "Health Pathogens"), c("FluB Pathogens", "Health Pathogens"), c("FluAB Pathogens", "Health Pathogens"))

ggplot(grid, aes(y=GRiD, x=groups)) +
    geom_jitter(aes(color = groups), width=0.2, size=1, alpha = 0.9)+
    geom_bar(stat="summary", fun=mean, position="dodge", fill = NA, color = "black") +
    stat_summary(geom = "errorbar",fun.data = 'mean_sd', width = 0.2)+
    theme_classic2()+
    scale_color_manual(values= c("#824533", "grey", "#b79570", "grey", "#161645", "grey", "#2b6b8e", "grey"))+
    labs( y="Bacterial growth rate", x="")+
    stat_compare_means(comparisons = my_comparisons, paired = FALSE, color="black")+ 
    theme(legend.position='none')+ theme(axis.text.x = element_text(angle = 45, hjust = 1))



#########################################################################
## Figure 5B

library(psych)
library(corrplot)

grid <- read.csv("dat/Figure_5_grid.csv",header = T, sep = ',')
data1 <- subset(grid, species %in% names1)

sp <- data1$species[!duplicated(data1$species)]

species <- read.csv("dat/Figure_5_species.csv",header = T, sep = ',', row.names = 1)

data1 <- subset(grid, species %in% names1)
data2 <- grid[-c(which(grid$species %in% names1)),]
test1 <- dcast(data1,id~data1$species,value.var = 'GRiD')
rownames(test1) <- test1$id
test1 <- test1[,-1]
test1 =  test1[, apply(test1, 2, function(y) length(na.omit(y))>5)]
test2 <- dcast(data2,id~data2$species,value.var = 'GRiD')
rownames(test2) <- test2$id
test2 <- test2[,-1]
test2 =  test2[, apply(test2, 2, function(y) length(na.omit(y))>5)]

cor <- corr.test(test1, test2)
r <- cor$r
p <- cor$p

r[p > 0.05] = 0
r1 <- r[, which(colSums(abs(r)) > 0)]


col3 <- colorRampPalette(c("blue", "white", "red"))
corrplot(r1,col=col3(20),method='color',tl.pos='lt', tl.col='black', sig.level=c(.001, .01, .05), pch.cex= .9, pch.col='black', xpd = T, tl.srt = 45)


########################################################################
## Figure 5C


data1 <- subset(grid, species %in% sp)
grid1 <- data1[, c(3,8,4)]
widedata2 <- dcast(grid1, id~species )
rownames(widedata2) <- widedata2$id
widedata2 <- widedata2[,-1]

path <- read.csv("dat/Figure_5_pathways.csv",header = T, sep = ',', row.names = 1)

path <- as.data.frame(t(path))
widedata2 <- widedata2[rownames(path),]

library(psych)
library(corrplot)
cor <- corr.test(widedata2, path, method = "spearman")
r <- cor$r
p <- cor$p
r[p > 0.05 | abs(r) < 0.5] = 0
r1 <- r[, which(colSums(abs(r)) > 0)]

##  The correlation data was used to imported to Cytoscape, and the Cytoscape output file is "Figure 5C.cys" in "dat" file.
```